src/components/CVideoStream/CVideoStream.js

```diff
@@ -1,71 +1,133 @@
-import { isNil, isObject } from 'lodash';
+import { isNil, isObject, template, upperFirst } from 'lodash';
 import Element from '../Element';
 
-const getStreamType = (config) => {
-  const type = isNil(config.streamType) ? 'img' : config.streamType;
-  return type;
-};
-
-const getStaticStyle = (config) => {
-  let height = 'auto';
-  let paddingTop = 0;
+export default {
+  extends: Element,
+  data() {
+    return {
+      player: null,
+      playerHandler: null,
+    };
+  },
+  computed: {
+    dependencies() {
+      if (this.streamType === 'Jsmpeg') {
+        return {
+          files: ['https://jsmpeg.com/jsmpeg.min.js'],
+          global: 'JSMpeg',
+        };
+      }
 
-  if (config.aspectRatio !== 'auto' && !isNil(config.aspectRatio)) {
-    height = 0;
-    const ratioValue = config.aspectRatio.split(':');
-    paddingTop = `${(ratioValue[1] / ratioValue[0]) * 100}%`;
-  }
+      return null;
+    },
+    staticStyle() {
+      let height = 'auto';
+      let paddingTop = 0;
 
-  const style = {
-    position: 'relative',
-    overflow: 'hidden',
-    width: '100%',
-    height,
-    paddingTop,
-  };
+      if (this.config.aspectRatio !== 'auto' && !isNil(this.config.aspectRatio)) {
+        height = 0;
+        const ratioValue = this.config.aspectRatio.split(':');
+        paddingTop = `${(ratioValue[1] / ratioValue[0]) * 100}%`;
+      }
 
-  return style;
-};
+      const style = {
+        position: 'relative',
+        overflow: 'hidden',
+        width: '100%',
+        height,
+        paddingTop,
+      };
 
-export default {
-  extends: Element,
-  watch: {
-    dataSource: {
-      handler() {
-        this.loadData();
-      },
-      deep: true,
+      return style;
     },
-  },
-  computed: {
     streamValue() {
       if (this.items && this.items.length) {
-        return isObject(this.items[0]) ? this.items[0].url : this.items[0];
+        const isMasked = !isNil(this.config.valueMask);
+        const isValueObject = isObject(this.items[0]);
+        const value = isValueObject ? this.items[0].url : this.items[0];
+
+        return isMasked && isValueObject ? template(this.config.valueMask)(this.items[0]) : value;
       }
+
       return this.config.value;
     },
+    streamType() {
+      let type = isNil(this.config.streamType) ? 'image' : this.config.streamType;
+
+      if (type === 'img') type = 'image';
+      if (type === 'mpeg') type = 'jsmpeg';
```

Comment:
If streamType is type 'select' inside options with 'image' and 'jmpeg' as values do we need this two conditions?

Reply:
No, I just wanted to be backward compatible for `img`. Will remove other. 👍